lib/era_ja/conversion.rb

```diff
@@ -32,7 +32,7 @@ def to_era(format = "%o%E.%m.%d", era_names: ERA_NAME_DEFAULTS)
       str_time = strftime(@era_format)
       if @era_format =~ /%([EOo]|1O)/
         case
-        when self.to_time < ::Time.mktime(1868,9,8)
+        when !era_convertible?
```

Comment:
I feel **a little** odd.
Because all conditions use `<` , but this condition calling method.
I would like to suggest the following code, what do you think?

```ruby
raise ERR_DATE_OUT_OF_RANGE unless era_convertible?
case
when self.to_time < ::Time.mktime(1912,7,30)
  str_time = era_year(year - 1867, :meiji, era_names)
...
```

Reply:
Indeed.  I moved this line to at the beginning of this method because ` era_convertible?` is not dependent on other variables defined inside this method.

https://github.com/tomiacannondale/era_ja/pull/19/files#diff-5b07a0ff388ba52729fcff09db4fc009R31